qiskit/wrapper/__init__.py

```diff
@@ -0,0 +1,151 @@
+        _DEFAULT_PROVIDER.add_provider(provider)
+    else:
+        raise QISKitError('provider name %s is not recognized' % provider_name)
+
+# Functions for inspecting and retrieving backends.
+
+
+def available_backends(filters=None):
+    """
+    Return the backends that are available in the SDK, optionally filtering
+    them based on their capabilities.
+
+    Note:
+        In order for this function to return online backends, a connection with
+        an online backend provider needs to be established by calling the
+        `register()` function.
+    Args:
+        filters (dict): dictionary of filtering conditions.
+    Returns:
+        list[str]: the names of the available backends.
+    """
+    return _DEFAULT_PROVIDER.available_backends(filters)
+
+
+def local_backends():
+    """
+    Return the available local backends.
+
+    Returns:
+        list[str]: the names of the available remote backends.
+    """
+    return available_backends({'local': True})
+
+
+def remote_backends():
+    """
+    Return the available remote backends.
+
+    Returns:
+        list[str]: the names of the available remote backends.
+    """
+    return available_backends({'local': False})
+
+
+def get_backend(name):
+    """
+    Return an instance of a `Backend` object from its identifier.
+
+    Args:
+        name(str): unique name of the backend.
+    Returns:
+        BaseBackend: a Backend instance.
+    """
+    return _DEFAULT_PROVIDER.get_backend(name)
+
+
+# Functions for compiling and executing.
+
+
+def compile(list_of_circuits, backend_id, compile_config=None):
+    """Compile a list of circuits into a qobj.
+
+    Args:
+        list_of_circuits (list[QuantumCircuits]): list of circuits
+        backend_id (str): a backend name to use as the default compiling option
+        compile_config (dict or None): a dictionary of compile configurations.
+            If `None`, the default compile configuration will be used.
+    Returns:
+        obj: the qobj to be run on the backends
+    """
+    # pylint: disable=redefined-builtin
+    backend = _DEFAULT_PROVIDER.get_backend(backend_id)
+    return qiskit._compiler.compile(list_of_circuits, backend, compile_config)
+
+
+def execute(list_of_circuits, backend_id, compile_config=None,
+            wait=5, timeout=60):
+    """Executes a set of circuits.
+
+    Args:
+        list_of_circuits (list[QuantumCircuits]): list of circuits.
+        backend_id (str): the name of the backend to execute the circuits on.
+        compile_config (dict or None): a dictionary of compile configurations.
+        wait (int): XXX -- I DONT THINK WE NEED TO KEEP THIS
+        timeout (int): XXX -- I DONT THINK WE NEED TO KEEP THIS
+
+    Returns:
+        Result: The results object
+    """
+    backend = _DEFAULT_PROVIDER.get_backend(backend_id)
+    return qiskit._compiler.execute(list_of_circuits, backend, compile_config,
+                                    wait, timeout)
```

Comment:
Is it common for this much code to go inside `__init__`, or should these functions be in a separate file inside the `wrapper` directory? I don't care that much, just curious if it is good style.

Reply:
It's common practice, but bordering the "acceptable number of lines limit" from my personal preference point of view! I'll just move it to a file, to pave the way for future additions.